hudi-client/src/main/java/org/apache/hudi/table/HoodieCommitArchiveLog.java

```diff
@@ -219,14 +220,29 @@ private boolean deleteArchivedInstants(List<HoodieInstant> archivedInstants) thr
    * @throws IOException in case of error
    */
   private boolean deleteAllInstantsOlderorEqualsInAuxMetaFolder(HoodieInstant thresholdInstant) throws IOException {
-    List<HoodieInstant> instants = metaClient.scanHoodieInstantsFromFileSystem(
-        new Path(metaClient.getMetaAuxiliaryPath()), HoodieActiveTimeline.VALID_EXTENSIONS_IN_ACTIVE_TIMELINE, false);
+    List<HoodieInstant> instants = null;
+    boolean success = true;
+    try {
+      instants =
+          metaClient.scanHoodieInstantsFromFileSystem(
+              new Path(metaClient.getMetaAuxiliaryPath()),
+              HoodieActiveTimeline.VALID_EXTENSIONS_IN_ACTIVE_TIMELINE,
+              false);
+    } catch (FileNotFoundException e) {
+      // On some FSs deletion of all files in the directory can auto remove the directory itself.
+      // GCS is one example, as it doesn't have real directories and subdirectories. When client
+      // removes all the files from a "folder" on GCS is has to create a special "/" to keep the folder
+      // around. If this doesn't happen (timeout, misconfigured client, ...) folder will be deleted and
+      // in this case we should not break when aux folder is not found.
+      // GCS information: (https://cloud.google.com/storage/docs/gsutil/addlhelp/HowSubdirectoriesWork)
```

Comment:
Guess it would be better to use multi line comments here like 
/*
*
*/

Reply:
updated